src/OpenTelemetry/Metrics/StringArrayEqualityComparer.cs

```diff
@@ -0,0 +1,69 @@
+// <copyright file="StringArrayEqualityComparer.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+
+namespace OpenTelemetry.Metrics
+{
+    internal class StringArrayEqualityComparer : IEqualityComparer<string[]>
```

Comment:
@alanwest I just took a quick look, I think there might be some runtime bits we can use for these array comparers. Check out this fiddle:

```csharp
using System;
using System.Collections;
using System.Collections.Generic;
					
public class Program
{
	public static void Main()
	{
		string[] first = new string[] { "one", "two", "three" };
		string[] second = new string[] { "one", "two", "three" };
		string[] third = new string[] { "one", "two", "THREE" };
		
		Console.WriteLine(((IStructuralEquatable)first).Equals(second, StringComparer.Ordinal));
		Console.WriteLine(((IStructuralEquatable)first).Equals(third, StringComparer.Ordinal));
		Console.WriteLine(((IStructuralEquatable)first).Equals(third, StringComparer.OrdinalIgnoreCase));
		
		double[] firstD = new double[] { 0.18D, 1D, 0.99D };
		double[] secondD = new double[] { 0.18D, 1D, 0.99D };
		
		Console.WriteLine(((IStructuralEquatable)firstD).Equals(secondD, EqualityComparer<double>.Default));
	}
}
```

It looks like Array has this hidden (explicitly implemented) [IStructuralEquatable](https://docs.microsoft.com/en-us/dotnet/api/system.array.system-collections-istructuralequatable-equals?view=net-6.0) we might be able to use.

Reply:
Cool I like this. From looking at [reference source](https://referencesource.microsoft.com/#mscorlib/system/array.cs,156e066ecc4ccedf) looks like Array has implemented IStructuralEquatable for quite some time.

Reply:
I reverted my change to use IStructuralEquatable. Unit tests were failing only for net461. I haven't dug into why. I'd like to follow up with trying out IStructureEquatable in a separate PR because it definitely does simplify the code.